### ISSUE TITLE
Osx Sierra compiling issues

### ISSUE THREAD

**Issue by jeppius** [pierreguillot/Camomile#136]
MacOS 10.13.6. I cloned the repo. So, how should I proceed?

cd Dependencies/LibPdBuild/MacOSX && cmake .. -GXcode && cd ../../..
-- Configuring done
CMake Error at CMakeLists.txt:258 (add_library):
  No SOURCES given to target: libpdstatic

-- Build files have been written to: /Users/josep 1/dev/Camomile/Camomile/Dependencies/LibPdBuild/MacOSX

**Comment by pierreguillot** [comment]
Did you clone the project and its submodules? Is there any file in Dependencies/LibPd or Juce? 
Normally, you have to do 
```
git clone --recursive https://github.com/pierreguillot/Camomile.git
```
or if you already cloned the repo
```
git submodule update --init --recursive
```

Build informations are on the Readme [#Compilation](https://github.com/pierreguillot/Camomile#compilation)

**Comment by jeppius** [comment]
Thanks, the --recursive flag did the trick.
Now I have another problem...while trying to compile the plugins as described in https://github.com/pierreguillot/Camomile/wiki/How-to-generate-plugins I cannot find the camomile script. Where is it located?

camomile admin$ ./camomile
-bash: ./camomile: No such file or directory

**Comment by jeppius** [comment]
Thanks the --recursive flag did the trick!.
Now I try to compile the plugins as described in https://github.com/pierreguillot/Camomile/wiki/How-to-generate-plugins
first I cd to Camomile/Plugins and type ./camomile 
I get this output:

-e Camomile - Plugin Generator
./camomile: line 365: [: /Users/josep: binary operator expected
-e Camomile.vst is not available
./camomile: line 320: [: /Users/josep: binary operator expected
-e Camomile.vst3 is not available
./camomile: line 271: [: /Users/josep: binary operator expected
-e Camomile.component is not available
./camomile: line 190: [: /Users/josep: binary operator expected
-e CamomileLV2.dylib is not available
./camomile: line 365: [: /Users/josep: binary operator expected
-e CamomileFx.vst is not available
./camomile: line 320: [: /Users/josep: binary operator expected
-e CamomileFx.vst3 is not available
./camomile: line 271: [: /Users/josep: binary operator expected
-e CamomileFx.component is not available
./camomile: line 190: [: /Users/josep: binary operator expected
-e CamomileLV2.dylib is not available
./camomile: line 365: [: /Users/josep: binary operator expected
-e CamomileFx.vst is not available
./camomile: line 320: [: /Users/josep: binary operator expected
-e CamomileFx.vst3 is not available
./camomile: line 271: [: /Users/josep: binary operator expected
-e CamomileFx.component is not available
./camomile: line 190: [: /Users/josep: binary operator expected
-e CamomileLV2.dylib is not available
./camomile: line 365: [: /Users/josep: binary operator expected
-e Camomile.vst is not available
./camomile: line 320: [: /Users/josep: binary operator expected
-e Camomile.vst3 is not available
./camomile: line 271: [: /Users/josep: binary operator expected
-e Camomile.component is not available
./camomile: line 190: [: /Users/josep: binary operator expected
-e CamomileLV2.dylib is not available
./camomile: line 365: [: /Users/josep: binary operator expected
-e CamomileFx.vst is not available
./camomile: line 320: [: /Users/josep: binary operator expected
-e CamomileFx.vst3 is not available
./camomile: line 271: [: /Users/josep: binary operator expected
-e CamomileFx.component is not available
./camomile: line 190: [: /Users/josep: binary operator expected
-e CamomileLV2.dylib is not available

**Comment by pierreguillot** [comment]
Perhaps, the problem is that there is a space in your path... Does it work well with the distribution (the pre-built distributed on the release section)? Can you try the script from somewhere else, where there is no space in the path? Or can you add quotation marks `"` around the paths is the script (but it can be a bit long to do everywhere)?

**Comment by pierreguillot** [comment]
I think I fixed the problem. 
Can you try using this branch [v1.0.7-fix-script](https://github.com/pierreguillot/Camomile/tree/dev/v1.0.7-fix-script) and tell me if it works for you?

**Comment by pierreguillot** [comment]
I did more tests and the fix works, feel free to reopen the issue if needed.

**Comment by theslyprofessor** [comment]
Hi Pierre,
I tried to download the script v1.0.7-fix-script that you mentioned but the link is dead. I store my work on iCloud so this would be beneficial to me.

Many thanks,
Nakul

**Comment by pierreguillot** [comment]
I think I merged the branch dev/v1.0.7-fix-script into [dev/v1.0.7](https://github.com/pierreguillot/Camomile/tree/dev/v1.0.7)
